Comparison between empty Set and Iterable fails

**Which version of Kotest are you using** 5.8.1


When comparing an empty set to an empty list:

```
java.lang.AssertionError: Disallowed: Sets can only be compared to sets, unless both types provide a stable iteration order.
EmptySet does not provide a stable iteration order and was compared with EmptyList which is not a Set
```

Does an empty set not have a very stable iteration order for the sake of comparison (since it does not have any elements?)
Could be solved by adding "size==0" as a clause where we haved "size==1" https://github.com/kotest/kotest/blob/ebcbd0e22fee429ddafaf9af38794816d347800d/kotest-assertions/kotest-assertions-shared/src/jvmMain/kotlin/io/kotest/assertions/eq/isOrderedSet.kt

I am also a bit surprised that `shouldBe` allows for this at all, i.e. matches a set(a,b,c) when a list(a,b,c) is expected?

Edit: Same issue for arrow's NonEmptyList, which I believe also has stable iteration order? Could perhaps be included in the kotest-arrow package

--- a/kotest-assertions/kotest-assertions-shared/src/jvmMain/kotlin/io/kotest/assertions/eq/isOrderedSet.kt
+++ b/kotest-assertions/kotest-assertions-shared/src/jvmMain/kotlin/io/kotest/assertions/eq/isOrderedSet.kt
@@ -8,4 +8,4 @@ import java.util.SortedSet
 actual fun isOrderedSet(item: Iterable<*>) =
    item is LinkedHashSet ||
       item is SortedSet ||
-      (item is Set && item.size == 1)
+      (item is Set && item.size <= 1)

--- a/kotest-assertions/kotest-assertions-shared/src/jvmTest/kotlin/com/sksamuel/kotest/eq/IterableEqTest.kt
+++ b/kotest-assertions/kotest-assertions-shared/src/jvmTest/kotlin/com/sksamuel/kotest/eq/IterableEqTest.kt
@@ -2,6 +2,8 @@ package com.sksamuel.kotest.eq
 
 import io.kotest.assertions.assertSoftly
 import io.kotest.assertions.eq.IterableEq
+import io.kotest.assertions.shouldFail
+import io.kotest.assertions.throwables.shouldNotThrowAny
 import io.kotest.core.spec.style.FunSpec
 import io.kotest.inspectors.forAll
 import io.kotest.matchers.nulls.shouldBeNull
@@ -41,6 +43,20 @@ private val expectedPath = if (System.getProperty("os.name").lowercase().contain
 }
 
 class IterableEqTest : FunSpec({
+   test("Comparing empty set with other iterable should be ok") {
+      shouldNotThrowAny {
+         emptySet<Int>() shouldBe listOf()
+      }
+   }
+
+   test("Comparing empty set with other iterable should provide meaningful assertion error") {
+      shouldFail {
+         emptySet<Int>() shouldBe listOf(1)
+      }.message shouldBe """
+         Missing elements from index 0
+         expected:<[1]> but was:<[]>
+      """.trimIndent()
+   }
 
    test("should give null for two equal sets") {
       IterableEq.equals(setOf(1, 2, 3), setOf(2, 3, 1)).shouldBeNull()

